src/mscorlib/src/System/Attribute.cs

```diff
@@ -459,6 +459,9 @@ public static Attribute[] GetCustomAttributes(MemberInfo element, Type type, boo
             if (type == null)
                 throw new ArgumentNullException(nameof(type));
 
+            if (type.IsGenericTypeDefinition)
```

Comment:
The api you want is `Type.ContainsGenericParameters`. This will catch open types such as `G<int, T>` in addition to `G<>`.